jmdaemon/jmdaemon/message_channel.py

```diff
@@ -412,12 +414,28 @@ def on_welcome_trigger(self, mc):
             self.mc_status[mc] = 1
             if self.welcomed:
                 return
+
+            #Startup sequence:
+            #Since this trigger was called, at least one mchan is ready.
             #This way broadcasts orders or requests ONCE to ALL mchans
             #which are actually available.
-            if not any([x == 0 for x in self.mc_status.values()]):
-                if self.on_welcome:
-                    self.on_welcome()
-                self.welcomed = True
+
+            # Any mchans not ready yet? Wait up to 60s for them.
+            if any([x == 0 for x in self.mc_status.values()]):
+                log.info("Could not connect to *ALL* servers yet, waiting " +
+                          "up to 60 more seconds.")
+                if (not self.on_welcome_announce_id) and self.on_welcome:
+                    self.on_welcome_announce_id = reactor.callLater(60, self.on_welcome_setup_finished,)
```

Comment:
Maybe this 60 second default should be made configurable? I was actually thinking today about this problem, maybe to look at it, my timeout would have been 30 seconds. :)

Reply:
I want to keep the number of config settings low, but ofc it isn't unreasonable to have it configurable. As-is with 60s should be fine; I think lower values leave too likely the possibility of a Taker accidentally not receiving the full liquidity when they `!orderbook`. If you want to PR it being configurable, that's fine, we can discuss it.